### ISSUE TITLE
Pollution of the Realms (adpother) cause crash on world creation

### ISSUE THREAD

**Issue by KubiRemPL** [LolHens/mc-fluid-physics#34]
Hi, so as sad in description adpother mod crashes when trying to create world.

Mods used:
**fluidphysics-1.4.0+forge-1.16.4.jar
ScalableCatsForce-2.13.5-build-2-with-library.jar
AdPother-1.16.4-4.0.3.0-build.0116.jar
ForgeEndertech-1.16.4-7.1.0.0-build.0075.jar**

Crashlog https://pastebin.com/tgh8T2Qd

**Comment by LolHens** [comment]
Do you also have optifine installed? If so you might be running into this issue:
https://github.com/LolHens/mc-fluid-physics/issues/29

**Comment by KubiRemPL** [comment]
Nope, tests were done only with those 4 mods

**Comment by cosmicdan** [comment]
Howdy,

Horray for open source! I may take a look at this for you, though I do have a lot of tasks for my Modpack I definitely want to have this (amazing work, thank you!) with said pollution mod, plus a bunch of others (Create water wheels, hydroelectric dams!) make this a must-have for my Modpack ideas.

Maybe this can be quick fixed by excluding adpother's fluid blocks from physics because they have their own basic implementation already? Though I imagine the mod design goal is to get the physics working with all fluids, not just liquids?

Thanks again for the mod! I'll try getting the source into a Dev Env. soon and see what I can do :)

**Comment by LolHens** [comment]
I just tried to reproduce this issue but for some reason couldn't.
I used 1.16.5 with forge 36.1.18 with the exact mods and versions you listed above.
Are you using the mojang launcher or something else with another java runtime?
In any case this issue is probably related to #29 and I would really like to fix this but there are some coremod bytecode injection conflicts involved which make this really hard to debug.
@cosmicdan Hi! Thanks for joining in! Sadly I think that excluding adpother's fluid blocks would not fix this since the bytecode is broken after coremod transformation and the broken code gets executed even if all fluids are disabled.
There is a workaround explained in #29 which disables one of adpother's bytecode transformations but that will break some block ticking stuff in adpother so I don't really recommend it.

**Comment by cosmicdan** [comment]
Yeah I can't reproduce either. AdPother + Fluid Physics both work fine together in my pack too.

**Comment by KubiRemPL** [comment]
So yeah, you are right guys. Issue is in JVM itself (I am using GraalVM). On normal Hotspot it worked fine.

**Comment by LolHens** [comment]
Okay thank you for confirming! I will try to debug this but it might be a graal bug.
